### PR TITLE
Color Me Javascript Challenge

### PR DESCRIPTION
**Pull request by X0rD3v1L**
Title : Color Me

Code Stack : Vanilla JS

### CODE REVIEW

**Comment by sadanandpai** [comment]
image
is this tested

**Comment by X0rD3v1L** [comment]
I have added a check to load the js once the DOM content is loaded, Can you please check now?

image

**Comment by sadanandpai** [comment]
I can see the same problem still.
https://deploy-preview-479--frontend-mini-challenges.netlify.app/javascript/src/challenges/color-me/

File: apps/javascript/src/challenges/color-me/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+document.addEventListener('DOMContentLoaded', () => {
```
Comment (sadanandpai) [inline apps/javascript/src/challenges/color-me/index.js:1]: as we are using defer, do we need DOMContentLoaded?

File: apps/javascript/src/challenges/color-me/index.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  <body>
+    <div class="container">
+      <h1>3x3 Grid Color Me</h1>
+
+      <div class="grid-container" id="grid"></div>
+
+      <div class="button-container">
+        <input type="number" id="inputBox" min="1" max="9" placeholder="Enter" />
+        <button id="colorButton">Color Me</button>
+      </div>
```
Comment (sadanandpai) [inline apps/javascript/src/challenges/color-me/index.html:19]: This should be a form. The value should be taken when the **Enter** key is pressed

File: apps/javascript/src/challenges/color-me/index.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    </div>
+
+    <script src="index.js" defer></script>
```
Comment (sadanandpai) [inline apps/javascript/src/challenges/color-me/index.html:22]: move to head section

File: apps/javascript/src/challenges/color-me/index.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>3x3 Round Grid Color Me</title>
```
Comment (sadanandpai) [inline apps/javascript/src/challenges/color-me/index.html:6]: above tags not needed

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by X0rD3v1L** [comment]
> I can see the same problem still. https://deploy-preview-479--frontend-mini-challenges.netlify.app/javascript/src/challenges/color-me/

type="module" resolved the issue, and removed the defer as modules are automatically deferred.

**Pull request merged**